### PR TITLE
Rename string and pattern types

### PR DESCRIPTION
**Pull request by hellerve**
This PR addresses #195 by renaming `string` to `String` in C land and `pattern` to `Pattern`.

We should probably still change the parsing to require a lower-case character at the start, but I’m not totally sure how to best do that.

Cheers

### CODE REVIEW

**Comment by eriksvedang** [comment]
Excellent! Yeah, not sure what to do about forcing lower case in variable names. It becomes a bit tricky when interop:ing ... but perhaps it's a good constraint on user defined variables in a carp program.

File: src/Parsing.hs [hunk 1]
```diff
@@ -4,6 +4,7 @@ module Parsing (parse, validCharacters, balance) where
 
 import Text.Parsec ((<|>))
 import qualified Text.Parsec as Parsec
+import qualified Data.Char as Char
```
Comment (hellerve) [inline src/Parsing.hs]: I just realized I introduced this spurious import here, @eriksvedang! Sorry about that!

Comment (eriksvedang) [reply src/Parsing.hs]: Its not needed?

Comment (hellerve) [reply src/Parsing.hs]: No, I added it when trying to patch the parser for non-capitalized names (did not work out, though, for obvious reasons).

Comment (eriksvedang) [reply src/Parsing.hs]: OK, I’ll remove it then, cheers.

**Pull request merged**